src/csm_testing/tests/validate_iuf_manifest/__main__.py

```diff
@@ -0,0 +1,99 @@
+# MIT License
+#
+# (C) Copyright 2024 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+"""
+This script validates IUF manifest.
+"""
+
+import yaml
+from jsonschema import validate, ValidationError, SchemaError
+import sys
+import os
+
+# Constants
+SCHEMA_FILE = "/opt/cray/tests/install/ncn/scripts/iuf_schemas/iuf-manifest-schema.yaml"  # Path to your product manifest schema file
+
+# Custom exception for IUF Product Manifest validation errors
+class ProductManifestValidationError(Exception):
+    """Custom exception for product manifest validation errors."""
+    pass
+
+def load_yaml(file_path):
```

Comment:
This same function is also defined in the `sat_validate_bootprep` test, with the primary difference being the names of the Exceptions being raised. This is also a good candidate to move into a shared module. That module can define and raise its own Exception, which the tests can each handle as appropriate.